executor/sort.go

```diff
@@ -410,22 +410,26 @@ func (e *TopNExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	if e.Idx >= len(e.rowPtrs) {
 		return nil
 	}
-	for !req.IsFull() && e.Idx < len(e.rowPtrs) {
-		row := e.rowChunks.GetRow(e.rowPtrs[e.Idx])
-		// Be carefule, if inline projection occurs.
-		// TopN's schema may be not match child executor's output columns.
-		// We should extract only the required columns from child's executor.
-		// Do not do it on `loadChunksUntilTotalLimit` or `processChildChk`,
-		// cauz it may destroy the correctness of executor's `keyColumns`.
-		req.AppendRowByColIdxs(row, e.columnIdxsUsedByChild)
-		e.Idx++
+	if !req.IsFull() {
+		numToAppend := mathutil.Min(len(e.rowPtrs)-e.Idx, req.RequiredRows()-req.NumRows())
+		rows := make([]chunk.Row, numToAppend)
```

Comment:
The modification here seems to be unnecessary, as `rows` is unnecessary.

Reply:
@pingyu It's copy from conflict merge:
```
	if e.Idx >= len(e.rowPtrs) {
		return nil
	}
	if !req.IsFull() {
		numToAppend := mathutil.Min(len(e.rowPtrs)-e.Idx, req.RequiredRows()-req.NumRows())
		rows := make([]chunk.Row, numToAppend)
		for index := 0; index < numToAppend; index++ {
			rows[index] = e.rowChunks.GetRow(e.rowPtrs[e.Idx])
			e.Idx++
		}
		req.AppendRows(rows)
	}
```
I am not so sure why here look like that... 
`chunk.AppendRows` vs. loop of `chunk.AppendRow`
Has any different? If the answer is "NO", I also think `rows` variable should not be alloced.

Reply:
These codes came from #21057.

As the benchmark shows that it brings performance improvement, it would be better to keep the `chunk.AppendRows`.
But I have a question about the result of benchmark, and left a [comment](https://github.com/pingcap/tidb/pull/21057#issuecomment-738181129) there.

Besides, add "xxxByColIdxs" again make the codes quite redundant...

Reply:
@pingyu Any good idea?

Reply:
@pingyu I think the lowest cost&highest performance implementation is add `chunk.AppendRowsByCols` ...
We can not prune chunk at `executeTopN`, it is expensive cauz extra memory allocated.
The only way seems like change columns at `append chunk` phase

Reply:
OK. I can't find a solution better than add `chunk.AppendRowsByColIdxs` (and `chunk.AppendPartialRowsByColIdxs`)...

Please be aware that the `rows []Row`  in `TopNExec.Next` should be pre-allocated. Refer to this [discussion](https://github.com/pingcap/tidb/pull/21057#issuecomment-738557164).

Reply:
@pingyu Agree.

Reply:
@pingyu PTAL